pkg/metrics/statsd/azure/metrics_test.go

```diff
@@ -0,0 +1,32 @@
+package azure
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+
+	mock_metrics "github.com/Azure/ARO-RP/pkg/util/mocks/metrics"
+)
+
+func TestTracer(t *testing.T) {
+	gc := gomock.NewController(t)
+	m := mock_metrics.NewMockInterface(gc)
+	m.EXPECT().EmitFloat("client.azure.duration", gomock.Any(), gomock.Any())
+	m.EXPECT().EmitGauge("client.azure.count", int64(1), gomock.Any())
+	m.EXPECT().EmitGauge("client.azure.errors", int64(1), gomock.Any())
```

Comment:
validate the content of dims (the third argument) instead of using gomock.Any().
(gomock.Any() is OK for the second argument in the duration case)
(throughout PR)